src/stream/src/executor/managed_state/join/mod.rs

```diff
@@ -146,29 +136,47 @@ impl<K: HashKey, S: StateStore> JoinHashMap<K, S> {
         target_cap: usize,
         pk_indices: Vec<usize>,
         join_key_indices: Vec<usize>,
-        data_types: Vec<DataType>,
+        mut data_types: Vec<DataType>,
         keyspace: Keyspace<S>,
+        dist_key_indices: Option<Vec<usize>>,
     ) -> Self {
-        let pk_data_types = pk_indices
+        let join_key_data_types = join_key_indices
             .iter()
             .map(|idx| data_types[*idx].clone())
             .collect_vec();
-        let join_key_data_types = join_key_indices
+
+        // Put the degree to the last column of the table.
+        data_types.push(DataType::Int64);
+
+        let column_descs = data_types
             .iter()
-            .map(|idx| data_types[*idx].clone())
+            .enumerate()
+            .map(|(id, data_type)| ColumnDesc::unnamed(ColumnId::new(id as i32), data_type.clone()))
             .collect_vec();
+
+        let table_pk_indices = [join_key_indices, pk_indices.clone()].concat();
+
+        // Order type doesn't matter here. Arbitrarily choose one.
+        let order_types = vec![OrderType::Descending; table_pk_indices.len()];
```

Comment:
I'm not sure whether `Ascending` can be faster to serialize due to no need of toggling bits.